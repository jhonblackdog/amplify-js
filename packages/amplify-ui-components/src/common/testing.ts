export const pixelThreshold = 10;
