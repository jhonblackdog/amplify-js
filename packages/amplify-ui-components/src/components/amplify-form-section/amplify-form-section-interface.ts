interface BaseFormSectionProps {
  overrideStyle?: boolean;
}

export interface AmplifyFormSectionHeaderProps extends BaseFormSectionProps {
  headerText: string;
}

export interface AmplifyFormSectionFooterProps extends BaseFormSectionProps {
  submitButtonText: string;
}