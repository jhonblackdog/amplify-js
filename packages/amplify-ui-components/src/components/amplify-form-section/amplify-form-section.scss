.form-section-header {
  color: var(--amplify-section-header-color);
  margin-bottom: 24px;
  font-size: var(--amplify-primary-font-size);
  font-weight: 700;
}

.form-section-footer {
  font-family: var(--amplify-font-family);
  font-weight: 300;
  font-size: var(--amplify-secondary-font-size);
  color: var(--amplify-grey);
  display: flex;
  flex-direction: row-reverse;
  justify-content: space-between;
  align-items: baseline;
}
