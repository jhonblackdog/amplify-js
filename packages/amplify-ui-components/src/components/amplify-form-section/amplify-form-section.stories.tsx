import { storiesOf } from '@storybook/html';
import { text as textKnob } from '@storybook/addon-knobs';

const formSectionStories = storiesOf('amplify-form-section', module);

formSectionStories.add('default', () => {
  return `
  <amplify-form-section></amplify-form-section>`;
});

formSectionStories.add('custom button text', () => {
  const submitButtonText = textKnob('Submit button', 'Go');

  return `<amplify-form-section submit-button-text=${submitButtonText}></amplify-form-section>`;
});
