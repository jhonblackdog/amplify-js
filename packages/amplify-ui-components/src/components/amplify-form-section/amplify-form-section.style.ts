import { css } from 'emotion';

export const formSectionHeader = css`
  color: var(--section-header-color);
  margin-bottom: 24px;
  font-size: 18px;
  font-weight: 500;
`;

export const formSectionFooter = css`
  font-size: 14px;
  color: var(--grey);
  display: flex;
  flex-direction: row-reverse;
  align-items: flex-start;
`;