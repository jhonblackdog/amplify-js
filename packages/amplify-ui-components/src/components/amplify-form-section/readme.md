# amplify-form-section



<!-- Auto Generated Below -->


## Properties

| Property           | Attribute            | Description                                                           | Type                          | Default     |
| ------------------ | -------------------- | --------------------------------------------------------------------- | ----------------------------- | ----------- |
| `handleSubmit`     | --                   | (Required) Function called upon submission of form                    | `(inputEvent: Event) => void` | `undefined` |
| `headerText`       | `header-text`        | Used for form section header                                          | `string`                      | `'Amplify'` |
| `overrideStyle`    | `override-style`     | (Optional) Overrides default styling                                  | `boolean`                     | `false`     |
| `submitButtonText` | `submit-button-text` | (Optional) Used as a the default value within the default footer slot | `string`                      | `'Submit'`  |


## Dependencies

### Used by

 - [amplify-sign-in](../amplify-sign-in)

### Depends on

- [amplify-section](../amplify-section)
- [amplify-button](../amplify-button)

### Graph
```mermaid
graph TD;
  amplify-form-section --> amplify-section
  amplify-form-section --> amplify-button
  amplify-sign-in --> amplify-form-section
  style amplify-form-section fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
