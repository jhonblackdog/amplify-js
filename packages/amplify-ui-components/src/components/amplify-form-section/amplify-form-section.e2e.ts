import { newE2EPage } from '@stencil/core/testing';
import { pixelThreshold } from '../../common/testing';

describe('amplify-form-section', () => {
  it('should render amplify-form-section with a button text as `Submit`', async () => {
    const page = await newE2EPage();
    await page.setContent(`<amplify-form-section></amplify-form-section>`);

    const el = await page.find('amplify-form-section');
    expect(el).not.toBeNull();

    const screenshot = await page.compareScreenshot('amplify-form-section', { fullPage: true });
    expect(screenshot).toMatchScreenshot({ allowableMismatchedPixels: pixelThreshold });
  });

  it('should change `button label` to `Enter`', async () => {
    const page = await newE2EPage();
    await page.setContent(`<amplify-form-section></amplify-form-section>`);

    const formSectionEl = await page.find('amplify-form-section');
    expect(formSectionEl).not.toBeNull();

    formSectionEl.setProperty('button-label', 'Enter');

    await page.waitForChanges();

    const buttonLabel = await formSectionEl.getProperty('button-label');

    expect(buttonLabel).toEqual('Enter');
  });
});