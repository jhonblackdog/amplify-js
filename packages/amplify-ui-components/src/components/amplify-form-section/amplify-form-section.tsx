import { Component, FunctionalComponent as FC, Prop, h } from '@stencil/core';
import { styleNuker } from '../../common/helpers';
import { AMPLIFY_UI_PREFIX } from '../../common/constants';
import { formSectionHeader, formSectionFooter } from './amplify-form-section.style';
import { AmplifyFormSectionHeaderProps, AmplifyFormSectionFooterProps } from './amplify-form-section-interface';

const STATIC_SECTION_HEADER_CLASS_NAME =  `${AMPLIFY_UI_PREFIX}--section-header`;
const STATIC_FORM_SECTION_FOOTER_CLASS_NAME = `${AMPLIFY_UI_PREFIX}--section-footer`;


const AmplifyFormSectionHeader: FC<AmplifyFormSectionHeaderProps> = ({ headerText, overrideStyle = false }) => (
  <div>
    <slot name="amplify-form-section-header">
      <div class={styleNuker(overrideStyle, STATIC_SECTION_HEADER_CLASS_NAME, formSectionHeader)}>
        <h2>{headerText}</h2>
      </div>
    </slot>
  </div>
);

const AmplifyFormSectionFooter: FC<AmplifyFormSectionFooterProps> = ({ submitButtonText, overrideStyle = false }) => (
  <div>
    <slot name="amplify-form-section-footer">
      <div class={styleNuker(overrideStyle, STATIC_FORM_SECTION_FOOTER_CLASS_NAME, formSectionFooter)}>
        <amplify-button type="submit">{submitButtonText}</amplify-button>
      </div>
    </slot>
  </div>
);

@Component({
  tag: 'amplify-form-section',
  shadow: false,
})
export class AmplifyFormSection {
  /** (Required) Function called upon submission of form */
  @Prop() handleSubmit: (inputEvent: Event) => void;
  /** (Optional) Used as a the default value within the default footer slot */
  @Prop() submitButtonText?: string = 'Submit';
  /** Used for form section header */
  @Prop() headerText: string = 'Amplify';
  /** (Optional) Overrides default styling */
  @Prop() overrideStyle?: boolean = false;

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <amplify-section override-style={this.overrideStyle}>
          <AmplifyFormSectionHeader headerText={this.headerText} overrideStyle={this.overrideStyle} />
          <slot />
          <AmplifyFormSectionFooter submitButtonText={this.submitButtonText} overrideStyle={this.overrideStyle} />
        </amplify-section>
      </form>
    );
  }
}