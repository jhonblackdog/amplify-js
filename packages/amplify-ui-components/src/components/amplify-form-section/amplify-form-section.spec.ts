import { newSpecPage } from '@stencil/core/testing';
import { AmplifyFormSection } from './amplify-form-section';

describe('amplify-form-section spec:', () => {
  describe('Component logic ->', () => {
    let formSection;

    beforeEach(() => {
      formSection = new AmplifyFormSection();
    });

    it('should render `Submit` for the button text', () => {
      expect(formSection.submitButtonText).toEqual('Submit');
    });

    it('`handleKeyDown` should be defined', () => {
      expect(formSection.handleKeyDown).toBeDefined();
    })

  });
  describe('Render logic ->', () => {
    it('should render a form section with a submit button text of `Go`', async () => {
      const page = await newSpecPage({
        components: [AmplifyFormSection],
        html: `<amplify-form-section submit-button-text="Go"></amplify-form-section>`,
      });

      expect(page.root).toMatchSnapshot();
    });

    it('should render a form section with a default button of `Submit`', async () => {
      const page = await newSpecPage({
        components: [AmplifyFormSection],
        html: `<amplify-form-section></amplify-form-section>`,
      });

      expect(page.root).toMatchSnapshot();
    });

    it('should render without Emotion CSS classes', async () => {
      const page = await newSpecPage({
        components: [AmplifyFormSection],
        html: `<amplify-form-section></amplify-form-section>`,
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});
