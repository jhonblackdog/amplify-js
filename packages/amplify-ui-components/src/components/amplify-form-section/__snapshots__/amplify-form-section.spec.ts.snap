// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-form-section spec: Render logic -> should render a form section with a default button of \`Submit\` 1`] = `
<amplify-form-section>
  <!---->
  <form>
    <amplify-section>
      <div>
        <slot-fb name="amplify-form-section-header">
          <div class="amplify-ui--section-header css-xdkp27">
            <h2>
              Amplify
            </h2>
          </div>
        </slot-fb>
      </div>
      <div>
        <slot-fb name="amplify-form-section-footer">
          <div class="amplify-ui--section-footer css-1q62vzk">
            <amplify-button type="submit">
              Submit
            </amplify-button>
          </div>
        </slot-fb>
      </div>
    </amplify-section>
  </form>
</amplify-form-section>
`;

exports[`amplify-form-section spec: Render logic -> should render a form section with a submit button text of \`Go\` 1`] = `
<amplify-form-section submit-button-text="Go">
  <!---->
  <form>
    <amplify-section>
      <div>
        <slot-fb name="amplify-form-section-header">
          <div class="amplify-ui--section-header css-xdkp27">
            <h2>
              Amplify
            </h2>
          </div>
        </slot-fb>
      </div>
      <div>
        <slot-fb name="amplify-form-section-footer">
          <div class="amplify-ui--section-footer css-1q62vzk">
            <amplify-button type="submit">
              Go
            </amplify-button>
          </div>
        </slot-fb>
      </div>
    </amplify-section>
  </form>
</amplify-form-section>
`;

exports[`amplify-form-section spec: Render logic -> should render a form section with only a class of \`amplify-ui--form-section\` when style override is true 1`] = `
<amplify-form-section override-style="true">
  <!---->
  <form>
    <amplify-section override-style="">
      <div>
        <slot-fb name="amplify-form-section-header">
          <div class="amplify-ui--section-header">
            <h2>
              Amplify
            </h2>
          </div>
        </slot-fb>
      </div>
      <div>
        <slot-fb name="amplify-form-section-footer">
          <div class="amplify-ui--section-footer">
            <amplify-button type="submit">
              Submit
            </amplify-button>
          </div>
        </slot-fb>
      </div>
    </amplify-section>
  </form>
</amplify-form-section>
`;
